### DISCUSSION TITLE
how can I put text inside node as children?

### DISCUSSION THREAD

**Discussion by X7md** [rehypejs/rehype#91] [Q&A] [answered]
[rehype figure](https://github.com/josestg/rehype-figure) don't work fine with me, so I had to cerate my own, anyway...

I need to put Text inside `<figcaption>`, as `TEXT_NODE`,

```
const replacement = {
        type: 'element',
        tagName: 'figure',
        properties: {},
        children: nodes.concat(
          [
            node,
            {
              type: 'element',
              tagName: 'figcaption',
              properties: {},
              /*Here... */
              children: nodes.concat({
                type: 'text',
                value: node.properties.alt
              })
            },
          ]
        )
      }
```

^ btw, that work but is it correct?

**Comment by ChristianMurphy** [comment, marked as answer]
Hey there @X7md 👋 
Sorry you ran into a spot of trouble.
Regarding running into an issue with `rehype-figure` it may be worth opening an [issue at the repo](https://github.com/josestg/rehype-figure/issues/new) with some additional information to let @josestg know what you're seeing that's unexpected.

In terms of is your replacement is correct or not, it's hard to tell with an incomplete snippet.
Sharing a full example of what you are thinking in a sandbox like [codesandbox](https://codesandbox.io/s/github/remarkjs/.github/tree/main/sandbox-templates/remark-rehype-with-parcel) or [stackblitz](https://stackblitz.com/github/remarkjs/.github/tree/main/sandbox-templates/remark-rehype-with-parcel) would help give folks the context needed to give a well rounded answer.
